File: app/src/debug/java/com/imangazalievm/bubbble/di/modules/OkHttpInterceptorsModule.java
package com.imangazalievm.bubbble.di.modules;

import androidx.annotation.NonNull;

import com.facebook.stetho.okhttp3.StethoInterceptor;
import com.imangazalievm.bubbble.di.qualifiers.OkHttpInterceptors;
import com.imangazalievm.bubbble.di.qualifiers.OkHttpNetworkInterceptors;

import java.util.List;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.Interceptor;
import okhttp3.logging.HttpLoggingInterceptor;

import static java.util.Collections.singletonList;

@Module
public class OkHttpInterceptorsModule {

    @Provides
    @OkHttpInterceptors
    @Singleton
    @NonNull
    public List<Interceptor> provideOkHttpInterceptors() {
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.HEADERS);
        return singletonList(httpLoggingInterceptor);
    }

    @Provides @OkHttpNetworkInterceptors
    @Singleton @NonNull
    public List<Interceptor> provideOkHttpNetworkInterceptors() {
        return singletonList(new StethoInterceptor());
    }

}

File: app-mvp/app/src/debug/java/com/bubbble/di/modules/OkHttpInterceptorsModule.kt
package com.bubbble.di.modules

import com.bubbble.di.qualifiers.OkHttpInterceptors
import okhttp3.logging.HttpLoggingInterceptor
import com.bubbble.di.qualifiers.OkHttpNetworkInterceptors
import com.facebook.stetho.okhttp3.StethoInterceptor
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.Interceptor
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class OkHttpInterceptorsModule {

    @Provides
    @OkHttpInterceptors
    @Singleton
    fun provideOkHttpInterceptors(): List<Interceptor> {
        val httpLoggingInterceptor = HttpLoggingInterceptor()
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.HEADERS)
        return listOf(httpLoggingInterceptor)
    }

    @Provides
    @OkHttpNetworkInterceptors
    @Singleton
    fun provideOkHttpNetworkInterceptors(): List<Interceptor> {
        return listOf(StethoInterceptor())
    }

}